packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts

```diff
@@ -107,4 +108,22 @@ export class R11sDocumentDeltaConnection extends DocumentDeltaConnection {
 			url: getUrlForTelemetry(this.url, socketIoPath),
 		};
 	}
+
+	/**
+	 * Disconnect from the websocket
+	 */
+	protected disconnectCore(err: IAnyDriverError): void {
+		// tell the server we are disconnecting this client from the document
+		const isCorruption =
+			err.errorType === FluidErrorTypes.dataCorruptionError ||
+			err.errorType === FluidErrorTypes.dataProcessingError;
+		this.socket.emit(
+			"disconnect_document",
+			this.clientId,
+			this.documentId,
+			err.errorType,
+			isCorruption,
```

Comment:
I think there should be a spec that defines the proposed contract between the client and sever, as we should really have specs when making protocol changes

Reply:
https://microsoft-my.sharepoint-df.com/:w:/p/rishhib/EeEPKFS4eIVNpAN11rXKWHgB4Rj1y4aa1I9pRgtYFkLeQQ?e=rSy0C7&nav=eyJjIjo0NTUyNjgyODd9